src/opm/parser/eclipse/EclipseState/Schedule/Well/Well.cpp

```diff
@@ -1376,6 +1382,9 @@ int Opm::eclipseControlMode(const Opm::Well::InjectorCMode imode,
 
     using IType = ::Opm::InjectorType;
 
+    if (wellStatus == ::Opm::Well::Status::SHUT) {
+        return Val::Shut;
+    }
```

Comment:
This is going to require updating [`WellTests`](https://github.com/OPM/opm-common/blob/269537b1015aedd904e55dc26ba18091bf25bf1d/tests/parser/WellTests.cpp#L927) too.